Feature/filter class coefficient setter

f-brinkmann opened the pull request:
### Which issue(s) are closed by this pull request?

Closes #249 

### Changes proposed in this pull request:

- add setter for `coefficients`.  Did not require new tests, because the setter is tested as part of the initialization.
- add `comment` property to all filter classes and new default for comment (`'None'` instead of `None`, which is more consistent with audio classes).

al-de reviewed (approved):
Thanks for implementing, this look good to me.

mberz commented:
See my comment in issue #249. I don't think having a setter like this is the way to go here. Let's discuss this further.

sikersten reviewed (commented):
Looks good, thanks! I'd generally approve, only two minor comments:

- Do I get it right, that the `coefficients` setter is only implicitly tested during the initialization and the use case of time variant filtering is more or less tested in filter_concepts.py?
- Maybe a little explanation of the intended use case in the documentation of the `coefficients` setter itself would be nice to hint to the problem mentioned by @mberz in #249?

f-brinkmann commented:
- Blockwise processing was tested for all filter calsses in `test_filter_classes.py::test_blockwise_processing`. I now added a test with exchanging the filter coefficients for the FIR filter `test_blockwise_processing_with_coefficients_exchange`. Together with the implicit tests of the setter functions this should cover all use cases.
- I added a link from  `Filter*.coefficients` to the Filter concepts.
